src/adapters/firebase.js

```diff
@@ -0,0 +1,109 @@
+const fs = require('fs')
+const https = require('https')
+const firebase = require('firebase/app')
+require('firebase/storage')
+global.XMLHttpRequest = require("xhr2")
+const path = require('path')
+
+const assertParam = require('../utils/assert-param')
+const resolve = require('./firebase-options');
+const { apiKey } = require('../../test/private/firebase_config');
+
+module.exports = config => {
+  const {context} = resolve(Object.assign(config, {validate:false}))
+  const appName = context.appName
+
+  let firebaseClient = {};
+  if (!firebase.apps.find(el => el.name_ == appName)) {
+    firebaseClient = firebase.initializeApp(context, appName)
+    console.log(`${appName} Firebase initialized.`)
+  } else {
+    firebaseClient = firebase.app(appName)
+  }
+
+  return {
+    /** 
+     * upload file on Firebase storage.
+     * @param file            File or blob object to be uploaded
+     * @param options.path        relative path for a file to be uploaded
+     * @param options.fileName      file name to be set 
+     * @param options.dispatch      function to dispatch events during this process 
+     * @param options.dispatchType    type defining progress of uploading a file
+     * @param options.afterUploaded   async function to execute 
+     */
+    upload (file, options) {
```

Comment:
minor: could you remove space between 'upload' and '('?
Or using `upload: function(file, options) {` style?

Reply:
I modified to 'update( '.